pex/resolve/resolver_options.py

```diff
@@ -135,8 +135,25 @@ def register(
             "See: https://pip.pypa.io/en/stable/topics/authentication/#keyring-support"
         ),
     )
+
     register_use_pip_config(parser)
 
+    parser.add_argument(
+        "--keyring-provider",
+        metavar="PROVIDER",
+        dest="keyring_provider",
+        type=str,
+        default=None,
+        help=(
+            "Configure Pip to use the given keyring provider to obtain authentication for package indexes. "
+            "You must be using Pip v23.1 or higher in order since keyring support was only added in Pip v23.1. "
```

Comment:
The `or higher in order since` bit needs some sort of re-wording fix.

Reply:
Reworded it to use "later versions" instead of "higher". Thoughts?

Reply:
The `or higher in order since` grammatical issue is gone so lgtm.